### PR TITLE
media: Add BufferInputDataSource

### PR DESCRIPTION
**Pull request by junmin-kim**
Add BufferInputDataSource

### CODE REVIEW

**Comment by junmin-kim** [comment]
@zhouxinhe Would you review this PR?

**Comment by seinfra** [comment]
Target : [3da03f5ac792d9e346ddc8884097a549986bb921] - Code Rule Check Result:
framework/include/media/BufferInputDataSource.h:42: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/include/media/BufferInputDataSource.h:70: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
framework/include/media/InputDataSource.h:82: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
framework/include/media/InputDataSource.h:121: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/InputDataSource.h:121: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:121: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:121: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:136: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/InputDataSource.h:136: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:136: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:136: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:137: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:137: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:137: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:142: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/InputDataSource.h:142: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:142: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:142: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxB)
framework/include/media/InputDataSource.h:143: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:143: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:143: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:144: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:144: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:144: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:146: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:146: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:146: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:153: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:153: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:153: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:154: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:154: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:154: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:155: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:155: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:155: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:156: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:156: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:156: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:157: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:157: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:157: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:163: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:164: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaPlayerObserverInterface.h:98: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaPlayerImpl.h:66: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/src/media/MediaPlayerImpl.h:66: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:66: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:66: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:66: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxE)
framework/src/media/MediaPlayerImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxB)
framework/src/media/MediaPlayerImpl.h:84: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:84: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:84: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:84: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxB)
framework/src/media/MediaPlayerImpl.h:103: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:103: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:103: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:103: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaPlayerImpl.h:104: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:104: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:104: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:104: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:104: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:104: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaPlayerImpl.h:108: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:108: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:108: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaPlayerImpl.h:111: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:112: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:113: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:113: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:113: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaPlayerImpl.h:114: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:114: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:114: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:114: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:114: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)

**Comment by seinfra** [comment]
Target : [3da03f5ac792d9e346ddc8884097a549986bb921] - Code Rule Check (C++) OK.

File: framework/include/media/MediaPlayerObserverInterface.h [hunk 1]
```diff
@@ -87,6 +87,15 @@ class MediaPlayerObserverInterface
 	 * @since TizenRT v2.0 PRE
 	 */
 	virtual void onPlaybackPaused(MediaPlayer &mediaPlayer) = 0;
+	/**
+	 * @brief informs the user that data in the input-buffer reach the
+	 * threshold, and it's possible to fetch data now.
+	 * @remark user need to override this method when the recorder working
```
Comment (Taejun-Kwon) [inline framework/include/media/MediaPlayerObserverInterface.h]: *recorder ??

**Comment by Taejun-Kwon** [comment]
More discussion needed.

File: framework/src/media/MediaPlayerImpl.cpp [hunk 2]
```diff
@@ -554,6 +554,14 @@ void MediaPlayerImpl::notifyObserver(player_observer_command_t cmd, ...)
         case PLAYER_OBSERVER_COMMAND_BUFFER_STATECHANGED:
```
Comment (JeongJunSik) [inline framework/src/media/MediaPlayerImpl.cpp]: please check tab

Comment (junmin-kim) [reply framework/src/media/MediaPlayerImpl.cpp]: Fixed it

File: framework/include/media/MediaPlayerObserverInterface.h (hunk 1 above)
Comment (junmin-kim) [reply framework/include/media/MediaPlayerObserverInterface.h]: Fixed it

**Comment by seinfra** [comment]
Target : [98d62ba27a8948f16552394f7fab2030e349fbac] - Code Rule Check Result:
framework/include/media/BufferInputDataSource.h:42: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/include/media/BufferInputDataSource.h:70: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
framework/include/media/InputDataSource.h:82: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
framework/include/media/InputDataSource.h:121: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/InputDataSource.h:121: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:121: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:121: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:136: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/InputDataSource.h:136: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:136: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:136: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:137: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:137: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:137: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:142: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/InputDataSource.h:142: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:142: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:142: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxB)
framework/include/media/InputDataSource.h:143: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:143: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:143: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:144: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:144: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:144: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:146: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:146: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:146: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:153: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:153: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:153: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:154: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:154: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:154: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:155: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:155: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:155: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:156: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:156: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:156: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:157: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:157: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:157: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:163: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:164: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaPlayerObserverInterface.h:98: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaPlayerImpl.h:66: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/src/media/MediaPlayerImpl.h:66: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:66: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:66: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:66: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxE)
framework/src/media/MediaPlayerImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxB)
framework/src/media/MediaPlayerImpl.h:84: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:84: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:84: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:84: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxB)
framework/src/media/MediaPlayerImpl.h:103: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:103: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:103: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:103: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaPlayerImpl.h:104: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:104: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:104: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:104: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:104: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:104: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaPlayerImpl.h:108: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:108: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:108: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaPlayerImpl.h:111: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:112: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:113: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:113: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:113: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaPlayerImpl.h:114: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:114: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:114: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:114: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:114: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)

**Comment by seinfra** [comment]
Target : [98d62ba27a8948f16552394f7fab2030e349fbac] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0b8493349d43adb9ed221342d1b54dd81dc774d5] - Code Rule Check Result:
framework/include/media/BufferInputDataSource.h:42: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/include/media/BufferInputDataSource.h:70: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
framework/include/media/InputDataSource.h:82: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
framework/include/media/InputDataSource.h:121: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/InputDataSource.h:121: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:121: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:121: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:142: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/InputDataSource.h:142: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:142: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:142: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxB)
framework/include/media/InputDataSource.h:143: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:143: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:143: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:144: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:144: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:144: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:146: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:146: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:146: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:153: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:153: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:153: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:154: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:154: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:154: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:155: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:155: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:155: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:156: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:156: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:156: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:157: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:157: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:157: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:163: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:164: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaPlayerObserverInterface.h:98: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaPlayerImpl.h:66: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/src/media/MediaPlayerImpl.h:66: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:66: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:66: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:66: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxE)
framework/src/media/MediaPlayerImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxB)
framework/src/media/MediaPlayerImpl.h:84: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:84: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:84: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:84: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxB)
framework/src/media/MediaPlayerImpl.h:103: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:103: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:103: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:103: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaPlayerImpl.h:104: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:104: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:104: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:104: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:104: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:104: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaPlayerImpl.h:108: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:108: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:108: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaPlayerImpl.h:111: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:112: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:113: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:113: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:113: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaPlayerImpl.h:114: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:114: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:114: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:114: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:114: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)

**Comment by seinfra** [comment]
Target : [0b8493349d43adb9ed221342d1b54dd81dc774d5] - Code Rule Check (C++) OK.

**Pull request merged**